src/NodaTime.Web/Program.cs

```diff
@@ -16,8 +39,77 @@ public static void Main(string[] args)
                 .UseIISIntegration()
                 .UseStartup<Startup>()
                 .Build();
+            host.Run(cancellationToken);
+        }
+
+        const int TestDurationSeconds = 15;
+        const int ServerStartupSeconds = 5;
+        const int GracefulShutdownSeconds = 5;
+        const string TestUrlBase = "http://localhost:5000";
+
+        private static async Task RunSmokeTests()
+        {
+            Environment.SetEnvironmentVariable("ASPNETCORE_ENVIRONMENT", "smoketests");
+            var cts = new CancellationTokenSource();
+            Task serverTask = Task.Run(() => Run(cts.Token));
+
+            // Give it 5 seconds to come up. We probably don't need this long, but it's harmless.
+            Thread.Sleep(ServerStartupSeconds * 1000);
 
-            host.Run();
+            bool success = false;
+            try
+            {
+                var testTask = FetchSmokePagesAsync();
+                var completed = await Task.WhenAny(testTask, Task.Delay(TimeSpan.FromSeconds(TestDurationSeconds)));
+                if (testTask != completed)
+                {
+                    throw new Exception($"Tests did not complete in {TestDurationSeconds} seconds");
+                }
+                await testTask;
+                success = true;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"Failure: {e.Message}");
+            }
+            // Try to let the server shut down gracefully.
+            cts.Cancel();
+            await Task.WhenAny(serverTask, Task.Delay(GracefulShutdownSeconds));
+
+            Console.WriteLine($"Smoke test complete. Result: {success}");
+            Environment.Exit(success ? 0 : 1);
+        }
+
+        private static async Task FetchSmokePagesAsync()
+        {
+            await ExpectPageText("downloads", "NodaTime-1.0.0.zip");
+            await ExpectPageText("1.0.x/userguide/text", "There are two options for text handling");
+            await ExpectPageText("1.0.x/api/NodaTime.DateTimeZone.html", "The mapping is unambiguous");
+            await ExpectPageText("tzdb/index.txt", "http://nodatime.org/tzdb/tzdb2013h.nzd");
+            await ExpectBinarySize("tzdb/tzdb2013h.nzd", 125962);
+        }
+
+        private static async Task ExpectPageText(string relativeUrl, string expectedText)
+        {
+            var client = new HttpClient();
+            string page = await client.GetStringAsync($"{TestUrlBase}/{relativeUrl}");
```

Comment:
Just checked, and we get a message of:

```text
Failure: Response status code does not indicate success: 404 (Not Found).
```

That doesn't show which URL was hit, but as the previous successes *are* reported, it's not too bad.